Tests/Creation/Frame/Content/String/ID3FrameFromURLStringContentCreatorTest.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ID3FrameFromStringContentCreatorTest.swift
+//
+//  Created by Nolaine Crusher on 03/31/2020.
+//  2018 Fabrizio Duroni.
+//
+
+import XCTest
+@testable import ID3TagEditor
+
+class ID3FrameFromURLStringContentCreatorTest: XCTestCase {
+    func testCreatorFrameFromURLStringContent() {
+        let id3FrameFromURLStringContentCreator = ID3FrameFromURLStringContentCreator(
+                frameContentSizeCalculator: MockFrameContentSizeCalculator(),
+                frameFlagsCreator: MockFrameFlagsCreator()
+        )
+
+        let frameBytes = id3FrameFromURLStringContentCreator.createFrame(
+                frameIdentifier: [0x22],
+                version: .version3,
+                content: "http://testurl.com"
+        )
+
+        XCTAssertEqual(frameBytes, [
+            0x22,//identifier
+            0x11, // size? How do I get this?
+            0x00, 0x01, // version?
+            0x00, 0x00, 0x68, 0x74, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x74, 0x65, 0x73, 0x74, 0x75, 0x72, 0x6c, 0x2e, 0x63, 0x6f, 0x6d, 0x00, 0x00 // http://testurl.com
+            ]
+        )
+    }
```

Comment:
Same here.

Reply:
Same here.

Why are you expecting a version?

From `0x68` to `0x6D` are the URL. Why are you expecting the two nulls in front of it and two nulls after it?

These little byte arrangement expectations come straight from the specification. It has a little table at the beginning of each frame describing how its bytes are arranged. (But the first part from the identifier to the flags is the same for all frames and is described in the introduction.)

Reply:
Honestly, I'm expecting a  version (and including those nulls at the beginning and end) because that seems to be the case in the other tests, and I'm doing my "monkey-see/monkey-do" thing just trying to imitate other examples without understanding them until it finally starts to click.

Specifically, I was mimicking the `ID3FrameFromStringContentCreator` test:

```swift
        let frameBytes = id3FrameFromStringContentCreator.createFrame(
                frameIdentifier: [0x22],
                version: .version3,
                content: "test"
        )

        XCTAssertEqual(frameBytes, [0x22, 0x11, 0x00, 0x01, 0x00, 0x00, 0x74, 0x65, 0x73, 0x74, 0x00, 0x00])
```

The `0x22` is the frameIdentifier. I'm not sure about the `0x11` byte. I assume `0x01` is the version (.version3 being the second of the three choices) and `0x74, 0x65, 0x73, 0x74` is "test" right? So what are the nulls at the beginning and end of it?

Reply:
It all comes straight from the specification, except that the various mocks are generating hogwash.

```
   All ID3v2 frames consists of one frame header followed by one or more
   fields containing the actual information. The header is always 10
   bytes and laid out as follows:

     Frame ID      $xx xx xx xx  (four characters)
     Size      4 * %0xxxxxxx
     Flags         $xx xx
```

- `0x22` is the identifier. (It ought to be four bytes long according to the specification, but the test is lazy and asks for a one‐byte identifier.)
- ` 0x11` is the size of the frame. (It ought to be four bytes long according to the specification, but `MockFrameContentSizeCalculator` is lazy and uses a one‐byte identifier.)
- `0x00` is the flag bytes. (Again `MockFrameFlagsCreator` is disobeying the specification and sending a single byte.)

```
4.2.   Text information frames

   The text information frames are often the most important frames,
   containing information like artist, album and more. There may only be
   one text information frame of its kind in an tag. All text
   information frames supports multiple strings, stored as a null
   separated list, where null is reperesented by the termination code
   for the charater encoding. All text frame identifiers begin with "T".
   Only text frame identifiers begin with "T", with the exception of the
   "TXXX" frame. All the text information frames have the following
   format:

     <Header for 'Text information frame', ID: "T000" - "TZZZ",
     excluding "TXXX" described in 4.2.6.>
     Text encoding                $xx
     Information                  <text string(s) according to encoding>
```

The rest comes straight from the `MockStringToBytesAdapter` below it, which completely ignores the specification and encodes those bytes no matter what string you give it.